icu4j/main/classes/core/src/com/ibm/icu/impl/PatternProps.java

```diff
@@ -119,6 +119,29 @@ public static String trimWhiteSpace(String s) {
         return s.substring(start, limit);
     }
 
+    /**
+     * @return s except with leading and trailing SpaceChar characters removed.
+     */
+    public static String trimSpaceChar(String s) {
```

Comment:
What sore of semantics do you want here?
This *looks* like it's unicode friendly except it's calling "isSpaceChar(char)" and not "isSpaceChar(int)":

https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isSpaceChar(int)

If you're only looking for ASCII this can be done far neater/smaller using "indexOf()"/"lastIndexOf()" and if you're trying to be Unicode friendly, use "isSpaceChar(int)".

Also, if performance isn't an issue in the case where

Reply:
Last sentence isn't complete?

Ah, surrogate pairs not supported, right?

* indexOf() / lastIndexOf() looks for one specific character, not sufficient.
* My very specific need is to trim space characters including the "NBSP": 
https://www.fileformat.info/info/unicode/char/00a0/index.htm - which is a SpaceChar but not Whitespace. isSpaceChar(char) seems adequate for my needs, but this should ideally behave consistently.

C++ code calls this:

```
U_CAPI UBool U_EXPORT2
u_isJavaSpaceChar(UChar32 c) {
    uint32_t props;
    GET_PROPS(c, props);
    return (UBool)((CAT_MASK(props)&U_GC_Z_MASK)!=0);
}
```

However the calling code is also just looking at one UChar at a time (16-bit). Here's my speculation now: maybe all space characters are within the Basic Multilingual Plane. So I'm tentatively asserting this code is Unicode friendly in the sense that it trims all Space Characters that are within the BMP?

In which case this change is unnecessary, but is the kind of code I think you're suggesting:

```
        if (s.length() == 0 || (!Character.isSpaceChar(s.codePointAt(0)) &&
                                !Character.isSpaceChar(s.codePointBefore(s.length())))) {
            return s;
        }
```

Reply:
Do you think it necessary to document that it trims only BMP space characters?

Reply:
Whitespace is all in the BMP.  If that ever changes, it will break a lot of assumptions throughout ICU and elsewhere.

Reply:
17 current "space separators" all in the BMP:
https://www.compart.com/en/unicode/category/Zs